src/ucp/core/ucp_worker.c

```diff
@@ -484,6 +484,27 @@ static ucp_ep_h ucp_worker_find_lane(ucs_list_link_t *ep_list, uct_ep_h uct_ep,
     return NULL;
 }
 
+/**
+ * FLUSH_CANCEL operation might be on pending queue due to
+ * UCS_ERR_NO_RESOURCES, so need to purge the queue to resubmit the
+ * operation. We need to resubmit the FLUSH_CANCEL operation on the same
+ * failed lane, in order to make sure all previous outstanding
+ * operations are completed before destroying the failed endpoint.
+ */
+static void ucp_worker_discard_lane_ff(ucp_worker_h worker, uct_ep_h uct_ep)
```

Comment:
1. why need worker arg?
2. consider just putting this code inside ucp_worker_iface_error_handler(): it's already pretty small and used only once

Reply:
> 1. why need worker arg?

leftover, thanks for catching!
> 2. consider just putting this code inside ucp_worker_iface_error_handler(): it's already pretty small and used only once

`ucp_worker_iface_error_handler` takes exactly 1 FHD screen in my current settings, putting this code inside including extra vars and comments will take half screen more, it does not look pretty small to me

Reply:
ok